src/transformers/generation/utils.py

```diff
@@ -1446,6 +1446,13 @@ def _supports_default_dynamic_cache(self) -> bool:
         """
         return self._supports_cache_class and "jamba" not in self.__class__.__name__.lower()
 
+    def _supports_num_logits_to_keep(self) -> bool:
+        """
+        Return True if the current model supports the keyword argument `num_logits_to_keep` in forward()
+        to save memory. Checking it in this way allows to avoid using a new model attribute.
+        """
+        return "num_logits_to_keep" in set(inspect.signature(self.forward).parameters.keys())
```

Comment:
Mmmmm I don't think we will add it to the signature, let's rather wait for this PR to be merged #31446 which will add kwargs

Reply:
I see the related PR only adding kwargs to the forward of the attention blocks, not to the forward of the main ModelForCausalLM. Will it be added in the future or am I missing something?

Reply:
Ignore what I said, as this is high impact feature, we can add it to the signature